src/network/networkd-dhcp-static-lease.c

```diff
@@ -0,0 +1,269 @@
+#include <errno.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#include "alloc-util.h"
+#include "hashmap.h"
+#include "networkd-util.h"
+#include "networkd-dhcp-static-lease.h"
+#include "memory-util.h"
+#include "strv.h"
+#include "utf8.h"
+
+DHCPStaticLease *dhcp_static_lease_free(DHCPStaticLease *static_lease) {
+        if (!static_lease)
+                return NULL;
+
+        if (static_lease->network) {
+                LIST_REMOVE(dhcp_static_leases, static_lease->network->dhcp_static_leases, static_lease);
+                assert(static_lease->network->n_dhcp_static_leases > 0);
+                static_lease->network->n_dhcp_static_leases--;
+
+                if (static_lease->section)
+                        hashmap_remove(static_lease->network->dhcp_static_leases_by_section, static_lease->section);
```

Comment:
I made it that way because the static leases have to be ordered and for it to be similar to other sections. If necessary I can remove the LIST and replace the `dhcp_static_leases_by_section` `hashmap` with an `ordered hashmap`. What are Your thoughts on that?